src/rust/ide/src/model.rs

```diff
@@ -1,24 +1,39 @@
 //! The Models.
 //!
 //! Models in this crate are structure which represents the logical "objects" being a part of the
-//! project, like project itself, module, execution context etc. They usually reflects the entites
+//! project, like project itself, module, execution context etc. They usually reflects the entities
 //! managed by LanguageServer.
 //!
-//! Some models are additionally wrapped in "Synchronized" structures, which adds an automatic
-//! synchronization between states of the model and the analogous entity in Language Server: for
-//! example `synchronized::Module` sends notification about each module change to LS.
+//! For Each model there is defined:
+//! * An API trait
+//! * A Synchronized Model implementing the API, which try to notify Language Server about all
+//!   changes in the modeled entity.
+//! * Sometimes a Plain Model, which stores data but does not synchronize anything.
+//! * Sometimes a Mock generated with `mock-all` crate.
+//! * A Shared Handle, which hides some of above API implementors under `Rc<dyn API>`.
 //!
-//! The models should implement internal mutability pattern, because they should be easily shared
-//! between controllers.
+//! Plain Model and Mock should be used in tests only, which simplifies mocking dependencies. In
+//! the actual code the Synchronized Model should be used. Each struct using model as a dependency
+//! should use the Shared Handle.
+//!
+//! Due to Shared Handle object, all models should implement "internal mutability" pattern.
```

Comment:
Let's rephrase that, like "As the models are usually shared between many controllers, they should implement "internal mutability" pattern."